src/net/server/handlers/create_query.rs

```diff
@@ -0,0 +1,439 @@
+                Request::post(format!(
+                    "/example?field_type={}",
+                    ff::Fp32BitPrime::TYPE_STR
+                ))
+                .body(ChunkedBody(chunks))
+                .unwrap(),
+            );
+            let mut byte_arr_stream = ByteArrStream::from_request(&mut req_parts).await.unwrap();
+            assert_eq!(byte_arr_stream.buffered.len(), 0);
+
+            // number of bytes pushed downstream
+            let mut num_downstream = 0;
+            // number of bytes pulled from upstream
+            let mut num_upstream =
+                u32::try_from(ARR_SIZE).unwrap() * ff::Fp32BitPrime::SIZE_IN_BYTES;
+            // expected size of buffer inside
+            let mut expected_size = 0;
+            for expected_n in vec.chunks(ff::Fp32BitPrime::SIZE_IN_BYTES as usize) {
+                // check that bytes are as expected
+                let n = byte_arr_stream.next().await.unwrap().unwrap();
+                assert_eq!(expected_n, &n);
+                assert!(byte_arr_stream.buffered.len() <= 1);
+
+                // sent Fp32BitPrime downstream
+                num_downstream += ff::Fp32BitPrime::SIZE_IN_BYTES;
+                // remove those bytes from buffer
+                expected_size -= i32::try_from(ff::Fp32BitPrime::SIZE_IN_BYTES).unwrap();
+                // if no more bytes in buffer, get more
+                if expected_size < 0 {
+                    // enough bytes upstream to pull a full chunk
+                    if CHUNK_SIZE < num_upstream {
+                        num_upstream -= CHUNK_SIZE;
+                        expected_size += i32::try_from(CHUNK_SIZE).unwrap();
+                    } else {
+                        // last chunk from upstream may be < full chunk size
+                        expected_size += i32::try_from(num_upstream).unwrap();
+                        num_upstream = 0;
+                    }
+                }
+                assert_eq!(
+                    byte_arr_stream.buffered_size,
+                    u32::try_from(expected_size).unwrap()
+                );
+            }
+            assert_eq!(
+                num_downstream,
+                u32::try_from(ARR_SIZE).unwrap() * ff::Fp32BitPrime::SIZE_IN_BYTES
+            );
+        }
+    }
+
+    mod prop_test {
+        use super::*;
+        use futures::TryStreamExt;
+        use proptest::prelude::*;
+
+        prop_compose! {
+            fn arb_aligned_bytes(size_in_bytes: u32, max_len: u32)
+                                (size_in_bytes in Just(size_in_bytes), len in 1..(max_len as usize))
+                                (vec in prop::collection::vec(any::<u8>(), len * size_in_bytes as usize))
+            -> Vec<u8> {
+                vec
+            }
+        }
+        prop_compose! {
+            fn arb_chunked_body(vec: Vec<u8>, field_type: &'static str)
+                               (chunk_size in 1..vec.len(), vec in Just(vec))
```

Comment:
need to find a way to get arbitrary-sized chunks. right now, all chunks are uniform in size

Reply:
ok, i managed to randomly slice the body in the proptest. I tried for a while to see if i could get it done using proptest `Strategy`s, but...that was proving really difficult and i think i would have to resort to implementing the trait myself, which looked like a lot of work.

so i just used `rng` with a seed, as you suggested